encoding/v2/kzg/prover/backend/icicle/multiframe_proof.go

```diff
@@ -119,21 +118,36 @@ func (p *KzgMultiProofBackend) ComputeMultiFrameProofV2(ctx context.Context, pol
 			}
 		}()
 
+		// Create a new stream for this operation to allow concurrent GPU operations
+		// without interference. Each stream can execute independently.
+		stream, streamErr := runtime.CreateStream()
+		if streamErr != runtime.Success {
+			icicleErr = fmt.Errorf("failed to create stream: %v", streamErr.AsString())
+			return
+		}
+		defer func() {
+			// Synchronize stream to ensure all GPU operations complete before cleanup
+			syncErr := runtime.SynchronizeStream(stream)
+			if syncErr != runtime.Success && icicleErr == nil {
+				icicleErr = fmt.Errorf("stream synchronization failed: %v", syncErr.AsString())
```

Comment:
do we really want to return an error if we couldnt synchronize the stream? Does it matter?